src/providers/jspm.ts

```diff
@@ -181,8 +183,28 @@ async function lookupRange (this: Resolver, registry: string, name: string, rang
     case 200:
       return { registry, name, version: (await res.text()).trim() };
     case 404:
+      const versions = await fetchVersions(name)
+      const semverRange = new SemverRange(range || '*', unstable)
+      const version = versions.find(v => semverRange.has(v))
+
+      if (version) {
+        return { registry, name, version };
+      }
       return null;
     default:
       throw new JspmError(`Invalid status code ${res.status} looking up "${registry}:${name}" - ${res.statusText}${importedFrom(parentUrl)}`);
   }
 }
+
+const versionsCacheMap = new Map<string, string[]>()
+
+async function fetchVersions(name: string): Promise<string[]> {
+  if (versionsCacheMap.has(name)) {
+    return versionsCacheMap.get(name);
+  }
+  const registryLookup = await (await fetch(`https://registry.npmjs.org/${name}`, {})).json();
```

Comment:
I believe this should fix the failing test?

```suggestion
  const registryLookup = await (await fetch(`https://npmlookup.jspm.io/${encodeURI(name)}`)).json();
```